### DISCUSSION TITLE
Report - (3rd Sem.) Week 1 📢

### DISCUSSION THREAD

**Discussion by maxschwinghammer** [SE-TINF22B6/DHBWhub#26] [Weekly Report]
Welcome to our first report of our project - DHBWhub! 👏 

### Let's introduce ourselves 🎉 
This is the team behind the project:
- Josha (@Nahlam4) - Product Owner
- Dennis (@denniskp) - Scrum Master
- Furkan (@FurKay00) - Developer
- Max (@maxschwinghammer) - Developer & Designer
- Jonas (@JonasEbbinghaus25) - Developer 
We have decided to not strictly assign developer roles and rather be more flexible and assign depending on the tasks! 

### What do we want to achieve with DHBWhub ❓ 
DHBWhub is going to be a website and Q&A-platform for students and other interested people, particularly in regards to 
the corporate state university DHBW and its courses. Students can discuss questions or talk about other things relating
to the organization of the university classes or other leisure activities. We want to start rather simple with pre-defined, 
stable structures and want to continuously add more features! 

![Logo of our website](https://github.com/SE-TINF22B6/DHBWhub/assets/111356450/851a99ff-df52-4776-bfba-7cde797093b2)

### Which platforms are we going to use? 🛠️ 
We intend to use these software platforms and applications for our project:
- React
- TypeScript
- Node.js
- Java
- JUnit
- Postgre SQL
- Spring Boot
- Maven
- SonarQube
(This list may change in the future...)

### How did we setup our Scrum? 📃 
- ✅Established [team agreement](https://github.com/SE-TINF22B6/DHBWhub/wiki/Team-Agreements)
- ✅Sprint length: 2 weeks
- ✅Main Scrum platform: GitHub and Discord for communication
- ✅Meetings:
   - Daily's on Monday and Thursday (roughly 15 min)
   - Weekly's on Wednesdays
   - Sprint Review, Retrospective, Planning and Refinement on every second Wednesday
- ✅Created task-hierarchy: Epic's - User story's - Tasks 
 
### Sprint 1: What have we achieved ❓  What's to come❔
- ✅Agreeing on the main idea for the project
- ✅Establishing Scrum-platform through GitHub
    - Sprint refinement
    - Categories
    - Templates for tasks
    - Corresponding labels
    - Detailed backlog
- ✅Agreeing on TechStack for now
- ✅Setting up React, Maven and Springboot
- ✅Integrate CI/CD with Dependabot
- ✅Create neat logo with no connotations to other websites
- 🚧Creating database and connecting with backend
- 🚧Using Figma for initial design for several pages
- 🚧Database schema for user accounts
- 🚧Discussing possible API's 
- 🚧Removing Herobrine 

### Lastly, what does our Kanban look like? 😮 
Currently, our board looks like this:
![grafik](https://github.com/SE-TINF22B6/DHBWhub/assets/122597204/a64619ae-57a4-4423-89f9-697ab20036a0)

**Comment by JAWolfs** [comment]
I like the overview of the roles and the motivation. I also think you got the most important tasks covered in the first sprint. 
What would be nice is a seperation of usages for the TechStack. What do you use in the frontend, what in the backend. That makes it easyer to get an overview for everybody that is not familiar with all of the software you use.
Last but not least I'm wondering if it's worth making a sprint appointment once a week. Do you work so consistently every week?
At least I think, we don't.

**Reply by maxschwinghammer** [reply]
@JAWolfs Since most of the people in our course know the technologies used, we do not think it is necessary to assign them to backend and frontend. 
In relation to your question: We work on our tasks every week, and therefore we like to have a weekly meeting by discussing thematic things as opposed to the Daily. However, please note that the Planning, Retro and Review meetings only take place every two weeks after the completion of a Sprint.

**Comment by tech-n1c** [comment]
Kudos for your SCRUM setup, looks like you already have a pretty clear vision for your project! We like that you've come up with a team agreement to have some guidelines that everyone should follow - this gives your project a solid structure and avoids bigger conflicts. As already mentioned, your SCRUM setup is already very detailed and you distributed all the important roles to your team members, thumbs up for that 👍 We are looking forward to seeing things develop at DHBWhub!
Sincereley, your team2shine 💫
